PowerEditor/src/ScintillaComponent/ScintillaEditView.cpp

```diff
@@ -646,6 +646,7 @@ void ScintillaEditView::setXmlLexer(LangType type)
 	}
 	else if ((type == L_HTML) || (type == L_PHP) || (type == L_ASP) || (type == L_JSP))
 	{
+		setLexerFromLangID(L_HTML);

```

Comment:
Could you also make the following change:

```cpp
void ScintillaEditView::setXmlLexer(LangType type)
{
	if (type == L_XML)
	{
		setLexerFromLangID(L_XML);
		for (int i = 0 ; i < 4 ; ++i)
```

Reply:
_Mais,regardez_ https://github.com/notepad-plus-plus/notepad-plus-plus/blob/3b0d5242acd23b4a727560ea98139cc5f4a8b88a/PowerEditor/src/ScintillaComponent/ScintillaEditView.cpp#L634-L641

_Alors_, `type = L_XML`:

~~~cpp
	setLexerFromLangID(L_XML);

	if (L_XML == L_XML)
	{
		for (int i = 0 ; i < 4 ; ++i)
			execute(SCISETKEYWORDS, i, reinterpretcast<LPARAM>(TEXT("")));
	// ...
	}
~~~

![npp-8-L_XML-detect](https://user-images.githubusercontent.com/59004801/165663743-e3fcbc4e-c33c-4f86-a35b-60ca68dd7f16.jpg)

_Problème? où est-il?_

[^1]: https://github.com/notepad-plus-plus/notepad-plus-plus/blob/a61b03ea8887e21c6e1b7374068962f635b79b80/PowerEditor/src/ScintillaComponent/ScintillaEditView.cpp#L631-L635

Reply:
What I want is make `setLexerFromLangID(L_XML);` inside of condition
```
if (type == L_XML)
{
```

so  `setLexerFromLangID` function won't be called twice in 
```cpp
else if ((type == L_HTML) || (type == L_PHP) || (type == L_ASP) || (type == L_JSP))
{
```

Reply:
_Bien entendu_: https://github.com/notepad-plus-plus/notepad-plus-plus/pull/11588/commits/cafafe1491a4e7a99489a246959e548da2b847e2

Reply:
_Cette [dernière révision][0] est de Visual Studio. De nos jours, c'est en l'UTF-8 que se passe_:

~~~diff
@@ -1426,7 +1427,7 @@ void ScintillaEditView::defineDocType(LangType typeDoc)
        pStyle = stylers.findByID(defaultIndicatorStyle._styleID);
        setSpecialIndicator(pStyle ? *pStyle : defaultIndicatorStyle);
 
-    // Il faut surtout faire un test ici avant d'ex<E9>cuter SCI_SETCODEPAGE
+    // Il faut surtout faire un test ici avant d'exécuter SCI_SETCODEPAGE
     // Sinon y'aura un soucis de performance!
        if (isCJK())
        {
~~~
[0]: https://github.com/notepad-plus-plus/notepad-plus-plus/commit/cafafe1491a4e7a99489a246959e548da2b847e2#diff-d88ddee57a027ab23daf332c4778ced0cee352edcb34efdda1b218e8a75c61b2R1430

Reply:
Merci @rdipardo !